include/beman/any_view/detail/iterator_adaptor.hpp

```diff
@@ -62,6 +62,30 @@ class iterator_adaptor final : public iterator_interface<ElementT, RefT, RValueR
         }
     }
 
+    [[nodiscard]] constexpr auto next() -> iter_cache_t<RefT> override {
+        if constexpr (std::forward_iterator<IteratorT>) {
+            return ++iterator != sentinel ? iter_cache_t<RefT>{*iterator} : std::nullopt;
+        }
+
+        unreachable();
```

Comment:
```suggestion
        std::unreachable();
```

maybe? not sure where this function is coming from

Reply:
`std::unreachable()` is C++23. This library has a polyfill so it can be implemented with C++20 support